rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -0,0 +1,414 @@
+        if (newFederationBtcUTXOs == null) {
+            return;
+        }
+
+        DataWord key = getStorageKeyForNewFederationBtcUtxos(networkParameters, activations);
+        bridgeStorageAccessor.safeSaveToRepository(key, newFederationBtcUTXOs, BridgeSerializationUtils::serializeUTXOList);
+    }
+
+    private void saveOldFederationBtcUTXOs() {
+        if (oldFederationBtcUTXOs == null) {
+            return;
+        }
+
+        bridgeStorageAccessor.safeSaveToRepository(OLD_FEDERATION_BTC_UTXOS_KEY.getKey(), oldFederationBtcUTXOs, BridgeSerializationUtils::serializeUTXOList);
+    }
+
+    private void saveNewFederation(ActivationConfig.ForBlock activations) {
+        if (newFederation == null) {
+            return;
+        }
+
+        if (!activations.isActive(RSKIP123)) {
+            bridgeStorageAccessor.safeSaveToRepository(NEW_FEDERATION_KEY.getKey(), newFederation, BridgeSerializationUtils::serializeFederationOnlyBtcKeys);
+            return;
+        }
+
+        bridgeStorageAccessor.safeSaveToRepository(NEW_FEDERATION_KEY.getKey(), newFederation, BridgeSerializationUtils::serializeFederation);
+        saveFederationFormatVersion(NEW_FEDERATION_FORMAT_VERSION.getKey(), newFederation.getFormatVersion());
+    }
+
+    private void saveOldFederation(ActivationConfig.ForBlock activations) {
+        if (!shouldSaveOldFederation) {
+            return;
+        }
+
+        if (!activations.isActive(RSKIP123)) {
+            bridgeStorageAccessor.safeSaveToRepository(OLD_FEDERATION_KEY.getKey(), oldFederation, BridgeSerializationUtils::serializeFederationOnlyBtcKeys);
+            return;
+        }
+
+        int oldFederationFormatVersion = getOldFederationFormatVersion();
+        bridgeStorageAccessor.safeSaveToRepository(OLD_FEDERATION_KEY.getKey(), oldFederation, BridgeSerializationUtils::serializeFederation);
+        saveFederationFormatVersion(OLD_FEDERATION_FORMAT_VERSION.getKey(), oldFederationFormatVersion);
+    }
+
+    private int getOldFederationFormatVersion() {
+        if (oldFederation == null) {
+            // assume it is a standard federation to keep backwards compatibility
+            return STANDARD_MULTISIG_FEDERATION.getFormatVersion();
+        }
+
+        return oldFederation.getFormatVersion();
+    }
+
+    private void savePendingFederation(ActivationConfig.ForBlock activations) {
+        if (!shouldSavePendingFederation) {
+            return;
+        }
+
+        byte[] serializedPendingFederation = serializePendingFederation(activations);
+        bridgeStorageAccessor.safeSaveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedPendingFederation);
+        if (!activations.isActive(RSKIP123)) {
+            return;
+        }
+
+        // we only need to save the standard part of the fed since the emergency part is constant
+        saveFederationFormatVersion(PENDING_FEDERATION_FORMAT_VERSION.getKey(), STANDARD_MULTISIG_FEDERATION.getFormatVersion());
```

Comment:
```suggestion
        if (activations.isActive(RSKIP123)) {
            // we only need to save the standard part of the fed since the emergency part is constant
        saveFederationFormatVersion(PENDING_FEDERATION_FORMAT_VERSION.getKey(), STANDARD_MULTISIG_FEDERATION.getFormatVersion());
        }
```
Easier to read

Reply:
Really? i thought we always wanted the current behavior last, and check for special scenarios

Reply:
And this is the current behaviour :) RSKIP123 is active

Reply:
yes! but i thought we wanted current behavior outside ifs / elses